v3/src/components/graph/utilities/graph-utils.ts

```diff
@@ -675,3 +679,49 @@ export const curveBasis = (points: Point[]) => {
   path += pathBasis(p2, p3, p3, p3)
   return path
 }
+
+const axisPlaceToOrientationMap: Record<GraphPlace, AxisOrientation> = {
+  "bottom": "horizontal",
+  "top": "horizontal",
+  "left": "vertical",
+  "rightCat": "vertical",
+  "rightNumeric": "vertical",
+  "yPlus": "vertical",
+  "plot": "horizontal",
+  "legend": "horizontal"
+}
+
+export interface IUpdateGraphValues {
+  attributeId: string | number
+  attributeName?: string
+  axisOrientation?: string
+  plotType?: string,
+  primaryRole?: string
+}
+
+export const attrChangeNotificationValues = (
+  place: GraphPlace, attrId: string, attrName?: string, attrIdToRemove = "",
+  tile?: ITileModel
+) => {
+
+  const axisOrientation = axisPlaceToOrientationMap[place] || "axis"
+
+  const attributeName = attrIdToRemove
+    ? `Remove ${ axisOrientation === "horizontal" ? "X" : "Y" }: ${attrName}`
+    : attrName
+
+  const graphContent = tile?.content as IGraphContentModel
+  const plotType = graphContent?.plotType
+  const primaryRole = graphContent?.dataConfiguration.primaryRole || "x"
```

Comment:
I'm not sure about whether it would be good to include this here or not. But I will say that in general, most plugins seem to ignore the content of notifications they receive, instead treating them as a signal that they need to send a `get` request to retrieve the information they care about.